app/controllers/documents_controller.rb

```diff
@@ -125,4 +134,37 @@ def prepare_for_mobile
     session[:mobile_param] = params[:mobile] if params[:mobile]
    # request.format = :mobile if mobile_device?
   end
+
+  private
+
+  def catalog_texts
+
+    if catalogue_enabled?
+       status, results = Melcatalog.texts
+       return results[:text] unless results[:text].nil?
+    end
+    return []
+  end
+
+  def catalog_content( doc )
+
+    if catalogue_enabled?
+      # we put placeholder content in earlier and replace with the real thing now
+      if doc.text.start_with?( "EID:" )
+         eid = doc.text.split( ":",2 )[ 1 ]
+         status, entry = Melcatalog.get( eid, 'stripxml' )
+         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
+           doc.text = entry[:text][ 0 ]['content']
+         else
+           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
+         end
+      end
+    end
+  end
+
+  # helper to determine if we should support content from the MEL catalog
+  def catalogue_enabled?
```

Comment:
Jamie,

I suppose we can either remove them from the application.yml or add a regex
filter that uses the CATALOG_ENABLED variable you use above to strip them
from the config string. I'm not sure which would be more in keeping with
your local idioms. I do hate multiple dependent configurations tho.

Thoughts?

Dave

On Mon, Feb 23, 2015 at 3:30 PM, Jamie Folsom <notifications@github.com>
wrote:

> In app/controllers/documents_controller.rb
> <https://github.com/hyperstudio/Annotation-Studio/pull/128#discussion_r25198781>
> :
>
> > +    if catalogue_enabled?
> > +      # we put placeholder content in earlier and replace with the real thing now
> > +      if doc.text.start_with?( "EID:" )
> > +         eid = doc.text.split( ":",2 )[ 1 ]
> > +         status, entry = Melcatalog.get( eid, 'stripxml' )
> > +         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
> > +           doc.text = entry[:text][ 0 ]['content']
> > +         else
> > +           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
> > +         end
> > +      end
> > +    end
> > +  end
> > +
> > +  # helper to determine if we should support content from the MEL catalog
> > +  def catalogue_enabled?
>
> I've added some config checking to this method, and when it is invoked, it
> does work.
>
>   # helper to determine if we should support content from the MEL catalog
>   def catalogue_enabled?
>     if ENV["CATALOG_ENABLED"] == true
>       # binding.pry
>       return true   # TODO: config...
>     end
>     # binding.pry
>     return false
>   end
>
> But I see that it's only invoked when you create a new document, and
> doesn't govern whether the features are enabled in the annotation editor.
>
> Ideally, we wouldn't see those buttons on the editor at all if this
> returns false. Since the buttons themselves are set in a config variable:
>
>   ANNOTATOR_RICHTEXT_CONFIG:              undo redo | styleselect | bold italic | link image media | code | mellink melimage
>
> I wonder if you have any thoughts as to how we should best handle the
> turning off of those buttons.
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hyperstudio/Annotation-Studio/pull/128/files#r25198781>
> .
>